crates/biome_js_analyze/src/lint/correctness/no_invalid_use_before_declaration.rs

```diff
@@ -105,20 +106,17 @@ impl Rule for NoInvalidUseBeforeDeclaration {
             let Ok(declaration_kind) = DeclarationKind::try_from(&declaration) else {
                 continue;
             };
-            let declaration_end = declaration.range().end();
-            let declaration_control_flow_root =
-                if let AnyJsBindingDeclaration::JsVariableDeclarator(declarator) = declaration
-                    .parent_binding_pattern_declaration()
-                    .unwrap_or(declaration)
-                {
-                    declarator
-                        .syntax()
-                        .ancestors()
-                        .skip(1)
-                        .find(|ancestor| AnyJsControlFlowRoot::can_cast(ancestor.kind()))
-                } else {
-                    None
-                };
```

Comment:
I don't know why we did this. No test seems to cover this. Previously we handled only variables and enum members. It seems that we wished to exclude enum members from same-scope checking.